frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -191,41 +271,68 @@ private void downloadLicenseInfo(ResourceRequest request, ResourceResponse respo
-            PortletResponseUtil.sendFile(request, response, filename, licenseInfoFile.getGeneratedOutput(), mimetype);
+            saveLicenseInfoAttachmentUsages(project, user, selectedReleaseAndAttachmentIds, excludedLicensesPerAttachmentId);
+            sendLicenseInfoResponse(request, response, project, licenseInfoFile);
         } catch (TException e) {
             log.error("Error getting LicenseInfo file for project with id " + projectId + " and generator " + outputGenerator, e);
             response.setProperty(ResourceResponse.HTTP_STATUS_CODE, Integer.toString(HttpServletResponse.SC_INTERNAL_SERVER_ERROR));
         }
     }
 
-    private void replaceAttachmentUsages(User user, Map<String, Set<String>> selectedReleaseAndAttachmentIds, Map<String, Set<LicenseNameWithText>> excludedLicensesPerAttachmentId, Project project) throws TException {
-        List<AttachmentUsage> attachmentUsages = ProjectPortletUtils.makeAttachmentUsages(project, selectedReleaseAndAttachmentIds,
-                excludedLicensesPerAttachmentId);
+    private void sendLicenseInfoResponse(ResourceRequest request, ResourceResponse response, Project project, LicenseInfoFile licenseInfoFile) throws IOException {
+        OutputFormatInfo outputFormatInfo = licenseInfoFile.getOutputFormatInfo();
+        String filename = String.format("LicenseInfo-%s-%s.%s", project.getName(), SW360Utils.getCreatedOn(),
+                outputFormatInfo.getFileExtension());
+        String mimetype = outputFormatInfo.getMimeType();
+        if (isNullOrEmpty(mimetype)) {
+            mimetype = URLConnection.guessContentTypeFromName(filename);
+        }
+
+        PortletResponseUtil.sendFile(request, response, filename, licenseInfoFile.getGeneratedOutput(), mimetype);
+    }
+
+    private void saveLicenseInfoAttachmentUsages(Project project, User user, Map<String, Set<String>> selectedReleaseAndAttachmentIds, Map<String, Set<LicenseNameWithText>> excludedLicensesPerAttachmentId) {
+        try {
+
+            Function<String, UsageData> usageDataGenerator = attachmentContentId -> {
+                Set<String> licenseIds = CommonUtils.nullToEmptySet(excludedLicensesPerAttachmentId.get(attachmentContentId)).stream()
+                        .filter(LicenseNameWithText::isSetLicenseName)
+                        .map(LicenseNameWithText::getLicenseName)
+                        .collect(Collectors.toSet());
+                return UsageData.licenseInfo(new LicenseInfoUsage(licenseIds));
+            };
+            List<AttachmentUsage> attachmentUsages = ProjectPortletUtils.makeAttachmentUsages(project, selectedReleaseAndAttachmentIds,
+                    usageDataGenerator);
+            replaceAttachmentUsages(project, user, attachmentUsages, UsageData.licenseInfo(new LicenseInfoUsage(Collections.emptySet())));
+        } catch (TException e) {
+            // there's no need to abort the user's desired action just because the ancillary action of storing selection failed
```

Comment:
Is the user notified, that some of the configuration is lost?

Reply:
No. Saving the selection is considered a dispensable action.